conda_forge_tick/update_recipe/v1/build_number.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Any, Literal
+
+from conda_forge_tick.update_recipe.v1.yaml import _dump_yaml_to_str, _load_yaml
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+HashType = Literal["md5", "sha256"]
+
+
+def _update_build_number_in_context(
+    recipe: dict[str, Any], new_build_number: int
+) -> bool:
+    for key in recipe.get("context", {}):
+        if key.startswith("build_") or key == "build":
```

Comment:
Which specific `build_*` keys are you targeting here? We should use an allowlist to ensure we don't stomp on any key that starts with `build_*`.

Reply:
changed to only target `build` and `build_number`.

Reply:
I changed this.